### ISSUE TITLE
Data issues obtained by Sui API

### ISSUE THREAD

**Issue by Gessi12** [MystenLabs/sui#18834]
URL: https://docs.sui.io/sui-api-ref#sui_getnormalizedmovemodule

Describe the issue or request. Provide as much detail as possible. For issues, it is helpful to copy the specific section of the topic into the issue.

`{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "sui_getNormalizedMoveModule",
  "params": [
    "0xb58b33843a07c2a7cf2707ca3ebf00d6de2ae1e5a273172c3615e9fd76d7e186",
    "router"
  ]
}`

I am using this API, but there is an issue with the data

![8c5bd1ba0e34dc4fc9e56c292b98db8](https://github.com/user-attachments/assets/df95b5c1-79e7-4f40-b1ca-71748f4cee8a)

The contract(0x2eeaab737b37137b94bfa8f841f92e36a153641119da3456dec1926b9960d9be) does not have a module named 'router'

![image](https://github.com/user-attachments/assets/d74c09c9-6e0c-4b63-8e8f-f86aec5805b7)

I think this CalculatedRouterSwapResult should be a CalculatedRouterSwapResult under the same contract

**Comment by stefan-mysten** [comment]
Thanks for the report @Gessi12!

**Comment by Gessi12** [comment]
> Thanks for the report @Gessi12!

I hope to resolve the issue as soon as possible and be informed that it has been resolved, so that users can obtain accurate data

**Comment by Eis-D-Z** [comment]
The package you mention: `0x2eeaab737b37137b94bfa8f841f92e36a153641119da3456dec1926b9960d9be` indeed does not have a router module.

The package you are querying though: `0xb58b33843a07c2a7cf2707ca3ebf00d6de2ae1e5a273172c3615e9fd76d7e186` has a router module. The RPC call works as intended in this case.

**Comment by Gessi12** [comment]
Yes, but I need the correct data, which is clearly incorrect @Eis-D-Z

**Comment by Eis-D-Z** [comment]
Yes I see it now. Sorry for that.
